File: iosApp/iosApp/Views/Components/ImagePicker.swift
import SwiftUI

struct ImagePicker: View {
    let onImageSelected: (String) -> Void
    let modifier: ViewModifier?

    init(
        onImageSelected: @escaping (String) -> Void = { _ in },
        modifier: ViewModifier? = nil
    ) {
        self.onImageSelected = onImageSelected
        self.modifier = modifier
    }

    var body: some View {
        Button(action: {
            // Handle image selection
            // This would typically open camera or gallery
            onImageSelected("selected_image_path")
        }) {
            VStack(spacing: 8) {
                Image(systemName: "photo")
                    .font(.system(size: 48))
                    .foregroundColor(.gray)

                Text("Add Image")
                    .font(.system(size: 16, weight: .medium))
                    .foregroundColor(.gray)
            }
            .frame(width: 200, height: 200)
            .background(Color.gray.opacity(0.2))
            .cornerRadius(16)
        }
    }
}

// Alternative version with frame modifier
extension ImagePicker {
    func frame(width: CGFloat, height: CGFloat) -> some View {
        Button(action: {
            onImageSelected("selected_image_path")
        }) {
            VStack(spacing: 8) {
                Image(systemName: "photo")
                    .font(.system(size: min(width, height) * 0.24))
                    .foregroundColor(.gray)

                Text("Add Image")
                    .font(.system(size: min(width, height) * 0.08, weight: .medium))
                    .foregroundColor(.gray)
            }
            .frame(width: width, height: height)
            .background(Color.gray.opacity(0.2))
            .cornerRadius(16)
        }
    }
}

File: androidApp/src/main/java/com/example/winemap/android/ui/components/ImagePicker.kt
@file:OptIn(ExperimentalMaterial3Api::class)

package com.example.winemap.android.ui.components

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun ImagePicker(
    onImageSelected: (String) -> Unit = {},
    modifier: Modifier = Modifier
) {
    Card(
        modifier = modifier.size(200.dp),
        shape = RoundedCornerShape(16.dp),
        colors = CardDefaults.cardColors(containerColor = Color.Gray.copy(alpha = 0.2f)),
        onClick = {
            // Handle image selection
            // This would typically open camera or gallery
            onImageSelected("selected_image_path")
        }
    ) {
        Column(
            modifier = Modifier.fillMaxSize(),
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.Center
        ) {
            Icon(
                painter = painterResource(id = android.R.drawable.ic_menu_gallery),
                contentDescription = "Add Image",
                tint = Color.Gray,
                modifier = Modifier.size(48.dp)
            )
            Spacer(modifier = Modifier.height(8.dp))
            Text(
                text = "Add Image",
                color = Color.Gray,
                fontSize = 16.sp,
                fontWeight = FontWeight.Medium
            )
        }
    }
}